### PR TITLE
Added churn_cal.py to machine_learning directory

### PR DESCRIPTION
**Pull request by IBR-41379**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by IBR-41379** [comment]
I hopefully followed all the rules. Still build is failing. Help me.

File: machine_learning/decision_tree_churn/churn_cal.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+def calculate_churn_rate() -> np.ndarray:
+    """
+    Calculates the churn rate of customers using a decision tree classifier.
+
+    >>> calculate_churn_rate()
+    array([0, 0, 0, 0, 1])
+    """
+
+    df = pd.read_csv("churn_modelling.csv")
```
Comment (cclauss) [inline machine_learning/decision_tree_churn/churn_cal.py:16]: https://docs.astral.sh/ruff/rules/pandas-df-variable-name

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/actions/runs/6485372101/job/17611303272?pr=10258#step:6:761

Scroll to line 760.

File: machine_learning/decision_tree_churn/churn_cal.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import doctest
```
Comment (cclauss) [inline machine_learning/decision_tree_churn/churn_cal.py:1]:
```suggestion
import doctest
from pathlib import Path
```

File: machine_learning/decision_tree_churn/churn_cal.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+def calculate_churn_rate() -> np.ndarray:
+    """
+    Calculates the churn rate of customers using a decision tree classifier.
+
+    >>> calculate_churn_rate()
+    array([0, 0, 0, 0, 1])
+    """
+
+    churn_md = pd.read_csv("churn_modelling.csv")
```
Comment (cclauss) [inline machine_learning/decision_tree_churn/churn_cal.py:16]: Enable experimentation with multiple files that are in the same directory as this file...

```suggestion
def calculate_churn_rate(filename: str = "churn_modelling.csv") -> np.ndarray:
    """
    Calculates the churn rate of customers using a decision tree classifier.

    >>> calculate_churn_rate()
    array([0, 0, 0, 0, 1])
    """

    churn_md = pd.read_csv(Path(__file__).with_name(filename).absolute())
```

**Review by cclauss — CHANGES_REQUESTED** [review]
You will need to add your .cvs file to this pull request.

File: machine_learning/decision_tree_churn/churn_cal.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    # Sorting the dependent and independent values
```
Comment (cclauss) [inline machine_learning/decision_tree_churn/churn_cal.py:18]: Use _active_ tense.
```suggestion
    # Sort the dependent and independent values
```

File: machine_learning/decision_tree_churn/churn_cal.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    # Creating the decision tree classifier
+    decision_tree = tree.DecisionTreeClassifier(criterion="entropy", max_depth=4)
+    decision_tree.fit(x_train, y_train)
+
+    # Predicting the values
+    pred_tree = decision_tree.predict(x_test)
+
+    # Returning the predicted values
+
+    return pred_tree[0:5]
```
Comment (cclauss) [inline machine_learning/decision_tree_churn/churn_cal.py:47]:
```suggestion
    # Split the dataset into training and testing data
    x_test, x_train, y_test, y_train = model_selection.train_test_split(
        x, y, test_size=0.3, random_state=3
    )

    # Create the decision tree classifier
    decision_tree = tree.DecisionTreeClassifier(criterion="entropy", max_depth=4)
    decision_tree.fit(x_train, y_train)

    # Returning the predicted values
    return decision_tree.predict(x_test)
```

File: machine_learning/decision_tree_churn/churn_cal.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+
+if __name__ == "__main__":
+    # Predicting the churn rate
+    pred_tree = calculate_churn_rate()
+    doctest.testmod()
+    # Printing the predicted values
+    print(pred_tree)
```
Comment (cclauss) [inline machine_learning/decision_tree_churn/churn_cal.py:55]: Run the tests _before_ doing calculations for the user.
```suggestion
    # Predicting the churn rate
    from doctest import testmod
    
    doctest.testmod()
    print(f"{calculate_churn_rate('churn_modelling.csv') = }")
```

**Comment by IBR-41379** [comment]
> You will need to add your .cvs file to this pull request.

I am pushing

**Comment by IBR-41379** [comment]
I am creting a new pull request as I hope it will solve the problem of csv file

**Pull request closed without merging**